src/goto-programs/json_goto_trace.cpp

```diff
@@ -114,6 +106,13 @@ void convert(
             type_string=from_type(ns, identifier, symbol->type);
 
           json_assignment["mode"]=json_stringt(id2string(symbol->mode));
+          assert(step.full_lhs_value.is_not_nil());
+          full_lhs_value=json(step.full_lhs_value, ns, symbol->mode);
+        }
+        else
+        {
+          if(step.full_lhs_value.is_not_nil())
```

Comment:
Actually the same occurs on several regression tests in cbmc: `Multi_Dimensional_Array6` `Initialization6`, `Pointer_byte_extract9` and `return5`